rio_stac/stac.py

```diff
@@ -0,0 +1,135 @@
+"""Create STAC Item from a rasterio dataset."""
+
+import datetime as pydatetime
+import os
+from contextlib import ExitStack
+from typing import Any, Dict, List, Optional, Sequence, Union
+
+import pystac
+import rasterio
+from rasterio import warp
+from rasterio.io import DatasetReader, DatasetWriter, MemoryFile
+from rasterio.vrt import WarpedVRT
+
+
+def bbox_to_geom(bbox: Sequence[float]) -> Dict:
+    """Return a geojson geometry from a bbox."""
+    # TODO: Handle dateline geometry
+    return {
+        "type": "Polygon",
+        "coordinates": [
+            [
+                [bbox[0], bbox[3]],
+                [bbox[0], bbox[1]],
+                [bbox[2], bbox[1]],
+                [bbox[2], bbox[3]],
+                [bbox[0], bbox[3]],
+            ]
+        ],
+    }
+
+
+def get_metadata(
+    source: Union[str, DatasetReader, DatasetWriter, WarpedVRT, MemoryFile]
+) -> Dict:
+    """Get Raster Metadata."""
+    metadata: Dict[str, Any] = {}
```

Comment:
Seems perfect for a dataclass... 😉 You've gotten me hooked!

Reply:
yes and no! Let's keep it like this for now. If we need to add more properties and method, maybe we will switch!

Reply:
Do we need to create our own `metadata` abstraction?  It might be easier to follow if we go straight from rasterio -> stac.

Reply:
@geospatial-jeff I'm not sure to understand